### PR TITLE
Remove permissions prop injection from main views

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: packages/ra-core/src/auth/useAuthenticated.ts [hunk 1]
```diff
@@ -28,9 +26,21 @@ const emptyParams = {};
...
-        checkAuth(params).catch(() => {});
-    }, [checkAuth, params]);
+        if (enabled) {
+            checkAuth(params).catch(() => {});
+        }
+    }, [checkAuth, enabled, params]);
+};
+
+export type UseAuthenticatedOptions<ParamsType> = {
+    enabled?: boolean;
```
Comment (WiXSL) [inline packages/ra-core/src/auth/useAuthenticated.ts:42]: It would be great to have documentation of the new `enabled` prop

File: packages/ra-core/src/auth/useCheckAuth.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ import useNotify from '../sideEffect/useNotify';
  *     return authenticated ? <Bar /> : <BarNotAuthenticated />;
  * } // tip: use useAuthState() hook instead
  */
-const useCheckAuth = (): CheckAuth => {
+export const useCheckAuth = (): CheckAuth => {
```
Comment (WiXSL) [inline packages/ra-core/src/auth/useCheckAuth.ts:43]: Missing export for type `CheckAuth`

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-core/src/auth/useAuthenticated.ts [hunk 3]
```diff
@@ -28,9 +26,21 @@ const emptyParams = {};
  *         </Admin>
  *     );
  */
-export default (params: any = emptyParams) => {
+export const useAuthenticated = <ParamsType = any>(
```
Comment (fzaninotto) [inline packages/ra-core/src/auth/useAuthenticated.ts:29]: this is a breaking change for users who used to pass params, and should be mentioned in the upgrade guide

File: packages/ra-core/src/controller/create/useCreateController.spec.tsx [hunk 4]
```diff
@@ -1,60 +1,49 @@
...
+    describe('getRecordFromLocation', () => {
+        const location: Location = {
+            key: 'a_key',
             pathname: '/foo',
             search: undefined,
             state: undefined,
             hash: undefined,
         };
 
-        it('should return an empty record by default', () => {
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/create/useCreateController.spec.tsx:20]: why did you remove this one ?

File: packages/ra-core/src/controller/create/useCreateController.ts [hunk 5]
```diff
@@ -49,20 +51,23 @@ export const useCreateController = <
     props: CreateControllerProps = {}
 ): CreateControllerResult<RecordType> => {
     const {
-        record = {},
-        successMessage,
+        disableAuthentication,
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/create/useCreateController.ts:54]: You should update the Auth doc to mention this prop. Same for other controllers.

**Review by fzaninotto — CHANGES_REQUESTED** [review]
The Resource component is becoming smaller and smaller... If we ever remove the registration phase, it'll just be routes!

File: packages/ra-core/src/controller/create/useCreateController.spec.tsx (hunk 4 above)
Comment (djhi) [reply packages/ra-core/src/controller/create/useCreateController.spec.tsx:20]: Because that's not the responsability of this function anymore

File: docs/Authentication.md [hunk 6]
```diff
@@ -701,11 +701,15 @@ const MyPage = () => {
 export default MyPage;
 ```
 
-If you call `useAuthenticated()` with a parameter, this parameter is passed to the `authProvider` call as second parameter. that allows you to add authentication logic depending on the context of the call:
+`useAuthenticated` accepts an options object as its only argument, with the following properties:
+- `enabled`: whether it should check for an authenticated user
```
Comment (WiXSL) [inline docs/Authentication.md:705]: - `enabled`: whether it should check for an authenticated user (`true` by default)

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/Authentication.md (hunk 6 above)
Comment (djhi) [reply docs/Authentication.md:705]: Make it a suggestion next time ;)
```suggestion
- `enabled`: whether it should check for an authenticated user (`true` by default)
```

**Review by WiXSL — APPROVED** [review]

File: docs/Authentication.md [hunk 7]
```diff
@@ -1165,3 +1169,30 @@ const Menu = ({ onMenuClick, logout }) => {
     );
 }
 ```
+
+### Allowing Anonymous Access to Custom Views
+
+You might have custom views that leverages our components or hooks such as:
```
Comment (fzaninotto) [inline docs/Authentication.md:1175]:
```suggestion
You might have custom views that leverage react-admin components or hooks such as:

```

**Review by fzaninotto — CHANGES_REQUESTED** [review]

**Review by fzaninotto — APPROVED** [review]
